sql/hive/src/test/scala/org/apache/spark/sql/hive/StatisticsSuite.scala

```diff
@@ -1484,4 +1484,41 @@ class StatisticsSuite extends StatisticsCollectionTestBase with TestHiveSingleto
       }
     }
   }
+
+  test("fallBackToHdfs should not support Hive partitioned table") {
+    Seq(true, false).foreach { fallBackToHdfs =>
+      withSQLConf(SQLConf.ENABLE_FALL_BACK_TO_HDFS_FOR_STATS.key -> s"$fallBackToHdfs",
+        HiveUtils.CONVERT_METASTORE_PARQUET.key -> "false") {
+        withTempDir { tableDir =>
+          withTable("spark_28876") {
+            sql(
+              s"""
+                 |CREATE TABLE spark_28876(id bigint)
+                 |PARTITIONED BY (ds STRING)
+                 |STORED AS PARQUET
+                 |LOCATION '${tableDir.toURI}'
+             """.stripMargin)
+            withTempDir { partitionDir =>
+              spark.range(5).write.mode(SaveMode.Overwrite).parquet(partitionDir.getCanonicalPath)
+              sql(s"ALTER TABLE spark_28876 ADD PARTITION (ds='p1') LOCATION '$partitionDir'")
+
+              val table = getCatalogTable("spark_28876")
+              assert(table.stats.isEmpty)
+              val partitions = spark.sessionState.catalog.listPartitions(table.identifier)
+              assert(partitions.size === 1)
+              partitions.foreach { p =>
+                // Partition directory is outside of the table directory
```

Comment:
we can simplify it a little
```
withTempDir { dir =>
  val tableDir = new File(dir, "table")
  val partitionDir = new File(dir, "partition")
}
```

Then we can guarantee that partition directory is outside of the table directory

Reply:
Done